stock_picking_group_by_partner_by_carrier/models/stock_picking.py

```diff
@@ -116,52 +115,53 @@ def copy(self, defaults=None):
         ).copy(defaults)
 
     def get_delivery_report_lines(self):
+        """Return the lines that will be on the report.
+
+        If the picking concerns multiple sale order some fake records are
+        inserted to have the sale information as line separators.
+        Otherwise standard records are returned.
+        """
         self.ensure_one()
+        moves = self.move_lines
         if self.state != "done":
-            moves = self.move_lines.filtered("product_uom_qty").sorted(
-                lambda m: m.sale_line_id.order_id
-            )
-            if len(moves.mapped("sale_line_id.order_id")) > 1:
-                sales_and_moves = []
-                for sale, sale_moves in groupby(
-                    moves, lambda m: m.sale_line_id.order_id
-                ):
-                    sales_and_moves.append(
-                        MockedMove(
-                            product_id=False,
-                            description_picking=sale.get_name_for_delivery_line(),
-                            product_uom_qty=0,
-                            product_uom=False,
-                            lot_name="",
-                        )
-                    )
+            moves = moves.filtered("product_uom_qty")
+        moves = moves.sorted(lambda m: m.sale_line_id.order_id)
+
+        if len(moves.mapped("sale_line_id.order_id")) > 1:
+            grouped_moves = groupby(moves, lambda m: m.sale_line_id.order_id)
+            fake_record = {
+                "product_id": 1,
+                "product_uom_qty": 0,
+                "product_uom": self.env.ref("uom.product_uom_unit").id,
+                "company_id": self.env.user.company_id.id,
+                "location_id": self.env.ref("stock.stock_location_output").id,
+                "location_dest_id": self.env.ref("stock.stock_location_output").id,
+            }
+            if self.state != "done":
+                sales_and_moves = self.env["stock.move"]
```

Comment:
this block of code seems a duplicate of the next one (beside 2 lines)... yeah, not blocking :stuck_out_tongue:

Reply:
True that.

Reply:
FYI, I did not improve the PR further on this. Because how it is now, it is (I think) clearer to see the both path of how the report lines are constructed.
And it may help later on for further improvement...